### ISSUE TITLE
Field "degrees" is required and must be assigned a value in constructor

### ISSUE THREAD

**Issue by hearot** [Microsoft/BosqueLanguage#69]
While I was trying to use Bosque, I ran into a problem and got this: `Field "degrees" is required and must be assigned a value in constructor`. The problem is: I'm already passing the `degrees` parameter!

I'm using Windows 10 and I used `node bin\test\app_runner.js angles.bsq` to launch the script.

<details>
<summary>Output</summary>

```
Reading app code...
Parse error -- ["angles.bsq",19,"Field \"degrees\" is required and must be assigned a value in constructor"]
```
</details>

<details>
<summary>Code - angles.bsq</summary>

```bosque
namespace NSMain;

entity Angle {
    field degrees: Int;
    field primes: Int = 0;
    field seconds: Int = 0;

    method normalized(over360?: Bool): Angle
        ensures _return_.primes < 60;
        ensures _return_.seconds < 60;
    {
        var! seconds = this.seconds % 60;
        var! primes = (this.primes + (this.seconds / 60)) % 60;
        var! degrees = this.degrees + (this.primes + (this.seconds / 60)) / 60;
        
        if (over360 && degrees > 360) {
            degrees = 360;
        }

        return Angle@{degrees, primes, seconds};
    }
}

entrypoint function main(): Angle {
    var angle = Angle@{degrees=0, primes=60, seconds=60};

    return angle->normalized(); // Output should be: Angle@{1, 1, 0}
}
```

</details>

**Comment by mrkmarron** [comment]
We currently require a bit more verbosity than is probably needed (just to be simple and unambigious to start) so the constructor call does not have name inferrence for the fields from the arguments. Thus, they need to be explicitly set. After this the code runs but there is a small odity in the value for degrees (probably a Int/Float related bug in Bosque).

```
return Angle@{degrees=degrees, primes=primes, seconds=seconds};
```

**Comment by mrkmarron** [comment]
FP/Int issue is fixed with PR #72 as well now.

**Comment by hearot** [comment]
Thank you, all is fixed now.
